app/models/classroom.rb

```diff
@@ -26,4 +26,11 @@ class Classroom < ApplicationRecord
   has_many :users, through: :student_classrooms
 
   validates :name, presence: true
+  validate :is_teacher, on: :create 
+ 
+  def is_teacher
+    if user.nil? || user.student?
```

Comment:
implement like this: `if user&.student?`
you can read more about this trick at http://mitrev.net/ruby/2015/11/13/the-operator-in-ruby/